### PR TITLE
Rsarwad handle 3485 timer

### LINKED ISSUE #3883: Handling timer 3485 and 3495 in stateless MME
When MME is in stateless mode, need to handle the following timers for registered UEs

1. 3485_timer: Purpose of this timer is, MME sends ACTIVATE DEDICATED EPS BEARER CONTEXT REQUEST to UE and waits for ACTIVATE DEDICATED EPS BEARER CONTEXT ACCEPT
On MME restart, MME deletes the context locally allocated for the bearer, if it's intended for all bearers, MME deletes all bearers and MME sends "Activation Bearer Reject" to SPGW
2. Timer_3495: Purpose of timer, MME sends DEACTIVATE EPS BEARER CONTEXT REQUEST and waits for DEACTIVATE EPS BEARER CONTEXT ACCEPT
On MME restart, MME shall deletes the context locally, send deactivation reject to SPGW

Ref: [Timer details document](https://docs.google.com/spreadsheets/d/1AUuwYfETPcswoVDNx53yyryLIFL3vjEagCEDhd_cu9I/edit#gid=1199317711)

### PR DESCRIPTION
**Pull request by rsarwad**
[AGW]: Added code to handle 3485 timer for both default and dedicated bearer while mme restarts

## Summary
1. Added code to handle 3485 timer for both default and dedicated bearer while UE is in EMM-Registered mode.
2. Added support to handle multi-pdn feature while MME restarts
3. Fixed some bugs and memory leaks found during the validation of 3485 timer for default bearer
4. During testing found that, some parameters/fields were not actually written or fetched from DB. Added these parameters to proto file and modified the code accordingly.

## Test Plan
Executed s1ap tester sanity test suite

## Additional Information
In order to test these changes, have dependency on PR raised on s1ap repo, https://github.com/facebookexperimental/S1APTester/pull/42

The test cases checked-in PR, https://github.com/magma/magma/pull/3970

### CODE REVIEW

**Comment by ssanadhya** [comment]
RestartIntegrationTests

**Comment by ssanadhya** [comment]
RestartLTETests

File: lte/gateway/c/oai/tasks/mme_app/mme_app_itti_messaging.c [hunk 1]
```diff
@@ -273,7 +273,8 @@ int mme_app_send_s11_create_session_req(
    * and will generate unique id only for 32 bits platforms.
    */
   OAI_GCC_DIAG_OFF("-Wpointer-to-int-cast");
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_itti_messaging.c]: Remove pointer to int cast,

File: lte/gateway/c/oai/tasks/mme_app/mme_app_itti_messaging.c [hunk 2]
```diff
@@ -273,7 +273,8 @@ int mme_app_send_s11_create_session_req(
...
-  session_request_p->sender_fteid_for_cp.teid = (teid_t) ue_mm_context;
+  session_request_p->sender_fteid_for_cp.teid =
+      (teid_t) ue_mm_context->mme_ue_s1ap_id;
   OAI_GCC_DIAG_ON("-Wpointer-to-int-cast");
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_itti_messaging.c]: Same as above

File: lte/gateway/c/oai/tasks/nas/esm/esm_ebr.c [hunk 3]
```diff
@@ -276,35 +278,33 @@ int esm_ebr_start_timer(
...
-       */
-      esm_ebr_timer_data->msg = bstrcpy(msg);
-
-      /*
-       * Setup the retransmission timer to expire at the given
-       * * * * time interval
-       */
-      ebr_ctx->timer.id =
-          nas_timer_start(sec, 0 /* usec */, cb, esm_ebr_timer_data);
+    if (ebr_ctx->args) {
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/nas/esm/esm_ebr.c:282]: Add comment indicating that, eventhough timer id is inactive but has timer arguments are non-null, indicates that data is read from Redis DB

File: lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp [hunk 4]
```diff
@@ -372,6 +407,7 @@ void NasStateConverter::proto_to_esm_context(
         esm_context_proto.esm_proc_data(), state_esm_context->esm_proc_data);
   }
   proto_to_nas_timer(esm_context_proto.t3489(), &state_esm_context->T3489);
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp:409]: Remove if not required

File: lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp [hunk 5]
```diff
@@ -391,6 +427,11 @@ void NasStateConverter::esm_ebr_context_to_proto(
   }
   nas_timer_to_proto(
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp:428]: Remove if not required

File: lte/gateway/c/oai/tasks/sgw/sgw_context_manager.c [hunk 6]
```diff
@@ -337,3 +335,54 @@ s_plus_p_gw_eps_bearer_context_information_t* sgw_cm_get_spgw_context(
...
+      (sgw_s11_teid_t*) calloc(1, sizeof(sgw_s11_teid_t));
+  if (sgw_s11_teid_p) {
+    sgw_s11_teid_p->sgw_s11_teid = teid;
+    LIST_INSERT_HEAD(&ue_context_p->sgw_s11_teid_list, sgw_s11_teid_p, entries);
+    OAILOG_DEBUG(
+        LOG_SPGW_APP,
+        "Inserted sgw_s11_teid to list of teids of UE context" TEID_FMT "\n",
+        teid);
+    OAILOG_FUNC_RETURN(LOG_SPGW_APP, RETURNok);
+  } else {
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/sgw/sgw_context_manager.c]: Make changes similar to line num, 366

File: lte/gateway/c/oai/tasks/sgw/spgw_state_converter.cpp [hunk 7]
```diff
@@ -955,15 +959,49 @@ void SpgwStateConverter::insert_proc_into_sgw_pending_procedures(
...
+            1, sizeof(s_plus_p_gw_eps_bearer_context_information_t)));
+    if (!spgw_context_p) {
+      OAILOG_DEBUG(
+          LOG_SPGW_APP, "Failed to allocate memory for SPGW context \n");
+      OAILOG_FUNC_OUT(LOG_SPGW_APP);
+    }
+
+    proto_to_spgw_bearer_context(S11BearerContext, spgw_context_p);
+    spgw_state_t* spgw_state       = get_spgw_state(false);
+    hash_table_ts_t* state_imsi_ht = get_spgw_ue_state();
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/sgw/spgw_state_converter.cpp]: change the name as state_ue_ht

File: lte/gateway/c/oai/tasks/sgw/spgw_state_converter.cpp [hunk 8]
```diff
@@ -955,15 +959,49 @@ void SpgwStateConverter::insert_proc_into_sgw_pending_procedures(
...
+        (s_plus_p_gw_eps_bearer_context_information_t*) (calloc(
+            1, sizeof(s_plus_p_gw_eps_bearer_context_information_t)));
+    if (!spgw_context_p) {
+      OAILOG_DEBUG(
+          LOG_SPGW_APP, "Failed to allocate memory for SPGW context \n");
+      OAILOG_FUNC_OUT(LOG_SPGW_APP);
+    }
+
+    proto_to_spgw_bearer_context(S11BearerContext, spgw_context_p);
+    spgw_state_t* spgw_state       = get_spgw_state(false);
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/sgw/spgw_state_converter.cpp]: Call these functions before for loop if size is non-zero

File: lte/protos/oai/nas_state.proto [hunk 9]
```diff
@@ -341,6 +341,13 @@ message ProtocolConfigurationOptions {
   repeated PcoProtocolOrContainerId proto_or_container_id = 5;
 }
 
+// 3485 and 3494 Timer argument
```
Comment (rsarwad) [inline lte/protos/oai/nas_state.proto]: Correct the timer as 3495

File: lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp (hunk 4 above)
Comment (rsarwad) [reply lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp:409]: This staement is for 3489 timer, which shall be removed while  handling clean-up of all timers

File: lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp (hunk 5 above)
Comment (rsarwad) [reply lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp:428]: same as above

**Review by ssanadhya — APPROVED** [review]

**Pull request merged**